### PR TITLE
OMIM source URL update

### PR DESCRIPTION
**Pull request by joeflack4**
> [!WARNING]
> Merging this also effectively merges #432, so I suggest to merge that first.

## Overview
OMIM release artefact now changed from `omim.ttl` to `omim.owl` ([context](https://github.com/monarch-initiative/omim/pull/107/files#r1479452702)).

## Pre-merge checklist

- [x] Docs
   - `docs/` have been added/updated **OR**
   - No updates to the docs necessary after careful consideration.

- [x] QC
   - `sh run.sh make build-mondo-ingest` has been run on this branch (after `docker pull obolibrary/odkfull:dev), and no errors occurred **OR**
   -  No functional (code-related) changes to the pipeline are suggested, so no re-run is necessary.

- [ ] Reviewed
   - Has been sufficiently reviewed by at least one review from a different team member of the Mondo Technical team.

---

CC @souzadevinicius

### CODE REVIEW

File: src/ontology/mondo-ingest-odk.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ components:
     - filename: ncit.owl
       source: http://purl.obolibrary.org/obo/ncit.owl
     - filename: omim.owl
-      source: https://github.com/monarch-initiative/omim/releases/latest/download/omim.ttl
+      source: https://github.com/monarch-initiative/omim/releases/latest/download/omim.owl
```
Comment (joeflack4) [inline src/ontology/mondo-ingest-odk.yaml:39]: ## The change
The only change other than the ones from `make update_repo`. See: [commit](https://github.com/monarch-initiative/mondo-ingest/pull/433/commits/7ec335f0cdf927b0ccf656e11b68a530247ddad4)

File: docs/odk-workflows/RepositoryFileStructure.md [hunk 2]
```diff
@@ -35,5 +35,5 @@ These are the components in MONDO-INGEST
 | icd10cm.owl | https://data.bioontology.org/ontologies/ICD10CM/submissions/23/download?apikey=8b5b7825-538d-40e0-9e9e-5ab9274a9aeb |
 | icd10who.owl | https://github.com/monarch-initiative/icd10who/releases/latest/download/icd10who.ttl |
 | ncit.owl | http://purl.obolibrary.org/obo/ncit.owl |
-| omim.owl | https://github.com/monarch-initiative/omim/releases/latest/download/omim.ttl |
+| omim.owl | https://github.com/monarch-initiative/omim/releases/latest/download/omim.owl |
```
Comment (joeflack4) [inline docs/odk-workflows/RepositoryFileStructure.md:38]: 3 changes made by `make update_repo`. See: [commit](https://github.com/monarch-initiative/mondo-ingest/pull/433/commits/ddc4ee006a20639f2159aa7f3130c79ef2ad8d0b)

File: src/ontology/Makefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # ----------------------------------------
 # Makefile for mondo-ingest
 # Generated using ontology-development-kit
-# ODK Version: v1.5
+# ODK Version: v1.5-dev
```
Comment (matentzn) [inline src/ontology/Makefile:4]: please use ODK_TAG=v1.4.3 as described on your ICD11 PR when running update_repo. There is too much spurious change in this PR.

**Review by matentzn — CHANGES_REQUESTED** [review]
Lets either merge #432 first or use the correct ODK Tag for update repo here

else ok

File: src/ontology/Makefile (hunk 3 above)
Comment (joeflack4) [reply src/ontology/Makefile:4]: > Lets either merge https://github.com/monarch-initiative/mondo-ingest/pull/432 first

Yes, I put in the OP that we should merge #432 first.

> please use ODK_TAG=v1.4.3
Yes, my bad. I'm having several issues regarding that, and the appropriate area to discuss that is here:
- https://github.com/monarch-initiative/mondo-ingest/pull/432#discussion_r1482008978

**Review by matentzn — APPROVED** [review]

**Review by matentzn — APPROVED** [review]
Ok, in the end, what happened was that because I was the only one updating the pipeline so far, the version of the pipeline corresponded to the ODK main branch - but that one is not regularly published (the `dev` branch has some more experimental features). 

However, as we are nearing an ODK release, and to enable you all (@twhetzel and @joeflack4 and @hrshdhgd) to run updates, lets keep using for now the `dev` image even for pipeline updates. There has not been a major ODK update for a while, but one is coming soon and it will look pretty much exactly like what we have now here in Mondo ingest. So, we are just a bit ahead of the curve.

In the future, we will resort to https://github.com/monarch-initiative/mondo-ingest/issues/435

**Pull request merged**